library/control/src/modules/WorkflowManager.rb

```diff
@@ -478,6 +467,46 @@ def addon_control_dir(src_id, cleanup: false)
       dir
     end
 
+    # Path of the control file contained in the package that has been previously
+    # extracted to the given directory
+    #
+    # @see #control_file
+    #
+    # @param dir [String] directory where the package has been extracted to
+    # @return [String] name of the control file
+    def control_file_at_dir(dir)
+      # Lets first try FHS compliant path for a product package (fate#325482)
+      path = find_control_file("#{dir}/usr/share/installation-products")
+
+      # If nothing there, try FHS compliant path for a role package (bsc#1114573)
+      path ||= find_control_file("#{dir}/usr/share/system-roles")
+
+      # As last resort, use the default location at /installation.xml
+      path ||= File.join(dir, "installation.xml")
+
+      path
+    end
+
+    # Full name of the control file located directly in the given directory
+    #
+    # The content of the file is not verified to be compliant with the structure
+    # of a control file, this method simply finds the (hopefully only) XML file
+    # in the directory.
+    #
+    # @param dir [String] directory where the control file is expected to be
+    # @return [String, nil] nil if there is no control file
+    def find_control_file(dir)
+      # sadly no glob escaping - https://bugs.ruby-lang.org/issues/8258
+      # but as we generate directory, it should be ok
+      files = Dir.glob("#{dir}/*.xml")
+
+      if files.size > 1
+        log.error "more than one file in #{dir}: #{files.inspect}"
```

Comment:
NP: I guess you are just keeping the log error as it was, but I would change it a little bit:

> **M**ore than one **XML** file in #{dir}: #{files.inspect}

At the end, it is only looking for _.xml_ files.